src/libsyntax/ext/quote.rs

```diff
@@ -646,6 +646,7 @@ fn expr_mk_token(cx: &ExtCtxt, sp: Span, tok: &token::Token) -> P<ast::Expr> {
 
         token::Literal(token::Byte(i), suf) => return mk_lit!("Byte", suf, i),
         token::Literal(token::Char(i), suf) => return mk_lit!("Char", suf, i),
+        token::Literal(token::Err(i), suf) => return mk_lit!("Err", suf, i),
```

Comment:
I think that here instead of calling `mk_lit` we might want to do something along the lines of `cx.expr(sp, ExprKind::Err)` (most likely) or `cx.ty(sp, TyKind::Err)`. That way we "forget" that this was a literal but only consider this as a placeholder that won't get type checked.